src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1416,7 +1417,11 @@ codet java_bytecode_convert_methodt::convert_instructions(
     else if(statement==patternt("?ipush"))
     {
       assert(results.size()==1);
-      results[0]=typecast_exprt(arg0, java_int_type());
+      namespacet ns(symbol_table);
+      // Force a simplify here in case we have --no-simplify
+      // because the simplifier is the only pass that knows how to
+      // deal with this cast from integer_typet:
+      results[0]=simplify_expr(typecast_exprt(arg0, java_int_type()), ns);
```

Comment:
Indeed. Either convert it back to a bitvector in the frontend (i.e. merge this PR) or have the parser create a bitvector to begin with (in which case I'll rewrite this Monday)